### PR TITLE
Add Youtube Channels endpoint

### PR DESCRIPTION
**Pull request by clloyd**
Adds a `/api/youtube/channels` list endpoint, also generalises the YoutubeCategories controller to be a general YouTube controller.

### CODE REVIEW

**Comment by clloyd** [comment]
@akash1810 added the safeguard we talked about in dev.

File: app/model/YouTubeChannel.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package model
+
+import com.google.api.services.youtube.model.Channel
+import play.api.libs.json._
+
+case class YouTubeChannel(
+                           name: String,
+                           logo: String,
```
Comment (akash1810) [inline app/model/YouTubeChannel.scala]: can this be a `URI` type?

File: app/util/Youtube.scala [hunk 2]
```diff
@@ -79,3 +80,21 @@ case class YouTubeVideoUpdateApi(config: YouTubeConfig) extends YouTubeBuilder {
       case _ => None
     }
 }
+
+case class YouTubeChannelsApi(config: YouTubeConfig) extends YouTubeBuilder {
+  def fetchMyChannels(): List[YouTubeChannel] = {
```
Comment (akash1810) [inline app/util/Youtube.scala]: very descriptive function name 😄

File: app/model/YouTubeChannel.scala (hunk 1 above)
Comment (akash1810) [inline app/model/YouTubeChannel.scala]: indents have gone a bit crazy

File: app/util/Youtube.scala [hunk 3]
```diff
@@ -20,6 +20,7 @@ class YouTubeConfig @Inject()(config: Configuration) {
   lazy val clientSecret = config.getString("youtube.clientSecret").getOrElse("")
   lazy val refreshToken = config.getString("youtube.refreshToken").getOrElse("")
   lazy val contentOwner = config.getString("youtube.contentOwner").getOrElse("")
+  lazy val allowedChannels = config.getStringList("youtube.allowedChannels")
```
Comment (akash1810) [inline app/util/Youtube.scala]: nb: we'd need to update the conf files in S3 before merging if we don't provide a default via `getOrElse`

Comment (clloyd) [inline app/util/Youtube.scala]: It's ok, it's used as an Option.

Comment (akash1810) [inline app/util/Youtube.scala]: would `youtube.devChannels` be more intention revealing?

Comment (clloyd) [inline app/util/Youtube.scala]: I like using a name that describes what it does... especially as they're not treated any differently than other channels. `devChannels` might imply other dev features?

Comment (akash1810) [inline app/util/Youtube.scala]: hmm, true.

**Comment by clloyd** [comment]
@akash1810 this ok now?

**Comment by akash1810** [comment]
@clloyd can we make `logo` a `URI` type? (the [original comment](https://github.com/guardian/media-atom-maker/pull/100#discussion_r89112697) isn't showing inline anymore for some reason...)

**Pull request merged**